src/core/connection.cpp

```diff
@@ -47,6 +50,10 @@ void Connection::stop_mavlink_receiver()
 
 void Connection::receive_message(mavlink_message_t& message, Connection* connection)
 {
+    // Register system ID when receiving a message from a new system.
+    if (_system_ids.find(message.sysid) == _system_ids.end()) {
```

Comment:
Out of curiosity: is it more costly to `find` first and `insert` if it does not exist, or to always `insert`? Given that `_system_ids` only has a few elements generally. Does it even make sense to think about that, or is it a detail?

Reply:
It probably does the same check inside, right? But if it already exists, we don't want to overwrite it, would be my thought.

Reply:
I am not really expert on this question, but what I understood is that finding over an `unordered_set` is very efficient, and as @julianoes said, `insert` might also calls to `find` function.

Reply:
I don't think that [insert](https://www.cplusplus.com/reference/unordered_set/unordered_set/insert/) overwrites it:

> Inserts new elements in the unordered_set.
> 
> Each element is inserted only if it is not equivalent to any other element already in the container (elements in an unordered_set have unique values).

That's why I was wondering if that test was not redundant.

Reply:
You are right, I got it know, indeed we could replace it if you prefer !